### PR TITLE
NUX: Add site preview flow and step config

### PR DESCRIPTION
**Pull request by ramonjd**
## What this PR does

Here we're integrating the site preview into the `onboarding-dev` flow (now _featuring the 'site style' step!_).

Borrowing from the work in #29400, this PR shifts the code that determines when to show the preview steps, and therefore the site mock up component, to the config.

jan-02-2019 17-21-28

## Testing
1. Head to http://calypso.localhost:3000/start/onboarding-dev
2. Navigate through the flow. Ensure that the site preview flow/style steps are displayed correctly.

This work does not effect the `onboarding` flow

### CODE REVIEW

File: client/signup/steps/site-type/index.jsx [hunk 1]
```diff
@@ -129,10 +129,11 @@ export default connect(
 			let nextFlowName = ownProps.flowName;
 			if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'store', 'slug' ) ) {
 				nextFlowName = 'ecommerce';
-			} else if ( 'ecommerce' === ownProps.flowName && ownProps.previousFlowName ) {
-				nextFlowName = ownProps.previousFlowName;
 			}
 
+			if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'business', 'slug' ) ) {
```
Comment (southp) [inline client/signup/steps/site-type/index.jsx]: Nitpick: this should be `else if`, otherwise this conditional statement will still be evaluated even if the previous one is true.

File: client/signup/steps/site-style/index.jsx [hunk 2]
```diff
@@ -124,15 +139,21 @@ export class SiteStyleStep extends Component {
...
 	}
 }
 
 const mapDispatchToProps = ( dispatch, ownProps ) => ( {
-	submitSiteStyle: ( siteStyle, themeSlugWithRepo ) => {
+	submitSiteStyle: ( siteStyle, themeSlugWithRepo, styleLabel ) => {
 		const { flowName, stepName, goToNextStep } = ownProps;
+		dispatch(
+			recordTracksEvent( 'calypso_signup_actions_submit_site_style', {
+				site_style: styleLabel,
```
Comment (southp) [inline client/signup/steps/site-style/index.jsx]: Any specific reason that the label is better than the `siteStyle` here?
As long as the site label won't be localized in the future and is as stable as the style id slug, I'm fine with keeping it this way.

Comment (ramonjd) [reply client/signup/steps/site-style/index.jsx]: Thanks for the review @southp !!!

> Any specific reason that the label is better than the siteStyle here?

Good question.  though it might be more descriptive as `siteStyle` values, such as 'modern' or 'default' are quite generic and might be used elsewhere. I might be wrong. 

The site label is the theme descriptor, so in the future it could be the name of the theme or other "product-like" label, which we don't translate. For example: https://it.wordpress.org/themes/

Having said that, I should probably make a comment to that effect. We're only using it for 4 (untranslated) strings atm.

File: client/signup/steps/site-type/index.jsx [hunk 3]
```diff
@@ -129,10 +129,9 @@ export default connect(
 			let nextFlowName = ownProps.flowName;
 			if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'store', 'slug' ) ) {
 				nextFlowName = 'ecommerce';
-			} else if ( 'ecommerce' === ownProps.flowName && ownProps.previousFlowName ) {
-				nextFlowName = ownProps.previousFlowName;
+			} else if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'business', 'slug' ) ) {
```
Comment (ramonjd) [inline client/signup/steps/site-type/index.jsx]: I just realized: this is the line that plugs it all in. 😄 Merging will effectively 'switch on' site style/preview in the flow for all users in the `onboarding` flow. 

Holding off on merging this PR until the final site preview designs have been implemented 

Since the preview styles are not yet finalized, we could revert this block and enable it only for `onboarding-dev` if we want to merge now, e.g.,

```
if ( siteTypeValue === getSiteTypePropertyValue( 'id', 'store', 'slug' ) ) {
				nextFlowName = 'ecommerce';
			} else if ( 'ecommerce' === ownProps.flowName && ownProps.previousFlowName ) {
				nextFlowName = ownProps.previousFlowName;
			} else if (  ( siteTypeValue === getSiteTypePropertyValue( 'id', 'business', 'slug' ) && 'onboarding-dev' === ownProps.flowName )  ) {
                ....
```
It might be useful to have to the updated flow config in dev to work with.

Or just put this PR on ice until the site preview is finalized. 

@michaeldcain

**Comment by scruffian** [comment]
Thanks for working on this.

File: client/signup/steps/site-type/index.jsx (hunk 3 above)
Comment (michaeldcain) [reply client/signup/steps/site-type/index.jsx]: > Since the preview styles are not yet finalized, we could revert this block and enable it only for onboarding-dev if we want to merge now

@ramonjd: sorry I missed the ping. Can we go ahead and launch this to just the onboarding-dev flow?

File: client/signup/site-mockup/index.jsx [hunk 4]
```diff
@@ -70,15 +70,7 @@ class SiteMockups extends Component {
 		return parts.slice( 0, 2 ).join( ', ' );
 	}
 
-	shouldRender() {
```
Comment (adamziel) [inline client/signup/site-mockup/index.jsx]: We only want to display the site preview for business sites. This could be a separate PR though to make the reviews smoother since we're only updating the `onboarding-dev` flow.

Comment (ramonjd) [reply client/signup/site-mockup/index.jsx]: > We only want to display the site preview for business sites. 

Thanks for the review! This PR enables such logic control but at a higher level in `flows-pure`. 

It does this by giving us the power to define flows _specifically_ for business. 

For example, when a user selects **Business** at the site style step, we can redirect to a flow that contains both the preview and the site style step.

It probably wouldn't hurt to be more defensive however so that, should a user land on this flow somehow with an undefined or non-Business site type, we shunt them back to the site type step. I'll explore in another PR, as you mentioned. 👍
